src/pybindings/pycontroller/BUILD.gn

```diff
@@ -0,0 +1,185 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import("//build_overrides/build.gni")
+import("//build_overrides/chip.gni")
+import("//build_overrides/pigweed.gni")
+
+import("$dir_pw_build/python.gni")
+
+import("${chip_root}/build/chip/tools.gni")
+import("${chip_root}/src/platform/device.gni")
+import("${dir_pw_unit_test}/test.gni")
+
+if (current_os == "mac") {
+  import("${build_root}/config/mac/mac_sdk.gni")
+}
+
+config("controller_wno_deprecate") {
+  cflags = [
+    "-Wno-deprecated-declarations",
+    "-Wno-shadow",
+    "-Wno-unused-result",
+    "-Wsign-compare",
+    "-Wunreachable-code",
+    "-Wno-macro-redefined",
+  ]
+}
+
+shared_library("CHIPController") {
+  configs -= [ "//build/config/compiler:exceptions_default" ]
+
+  output_name = "PyChip"
+  output_dir = "${target_out_dir}/pychip"
+  include_dirs = [ "${chip_root}/third_party/pybind11/repo/include" ]
+  if (current_os == "mac") {
+    include_dirs +=
```

Comment:
Can we use GN's `exec_script` to somehow determine the actual paths? I'm fine with changing that later, but you could add some "TODO:" not to forget about it.